yt_dlp/extractor/pbs.py

```diff
@@ -696,3 +698,73 @@ def extract_redirect_urls(info):
+    _TESTS = [
+        {
+            'url': 'https://pbskids.org/video/molly-of-denali/3030407927',
+            'md5': '1ded20a017cc6b53446238f1804ce4c7',
+            'info_dict': {
+                'id': '3030407927',
+                'title': 'Bird in the Hand/Bye-Bye Birdie',
+                'channel': 'molly-of-denali',
+                'duration': 1540,
+                'ext': 'mp4',
+                'series': 'Molly of Denali',
+                'description': 'md5:d006b2211633685d8ebc8d03b6d5611e',
+                'categories': ['Episode'],
+                'upload_date': '20190718',
+            }
+        },
+        {
+            'url': 'https://pbskids.org/video/plum-landing/2365205059',
+            'md5': '92e5d189851a64ae1d0237a965be71f5',
+            'info_dict': {
+                'id': '2365205059',
+                'title': 'Cooper\'s Favorite Place in Nature',
+                'channel': 'plum-landing',
+                'duration': 67,
+                'ext': 'mp4',
+                'series': 'Plum Landing',
+                'description': 'md5:657e5fc4356a84ead1c061eb280ff05d',
+                'categories': ['Episode'],
+                'upload_date': '20140302',
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        # Grabbing JSON block linked to window._PBS_KIDS_DEEPLINK
+        meta = json.loads(self._search_regex(r'window\._PBS_KIDS_DEEPLINK = (?P<video_info>\{.*\})', webpage, 'video_info'))
+
+        # Extract relevant bits
+        air_date = traverse_obj(meta, ('video_obj', 'air_date'), expected_type=str)
+        channel = meta.get('show_slug')
+        description = traverse_obj(meta, ('video_obj', 'description'), expected_type=str)
+        duration = traverse_obj(meta, ('video_obj', 'duration'), expected_type=int_or_none)
+        m3u8_playlist = traverse_obj(meta, ('video_obj', 'URI'), expected_type=url_or_none)
+        series = traverse_obj(meta, ('video_obj', 'program_title'), expected_type=str)
+        title = traverse_obj(meta, ('video_obj', 'title'), expected_type=str)
+        # Differentiate between clips and episodes
+        video_type = traverse_obj(meta, ('video_obj', 'video_type'), expected_type=str)
+        # expire_date = parse_iso8601(video_info["video_obj"]["expire_date"])
+        # Parse out formats
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(m3u8_playlist, video_id, ext='mp4')
+
+        return {
+            'categories': [video_type],
+            'channel': channel,
+            'description': description,
+            'duration': duration,
+            'formats': formats,
+            'id': video_id,
+            'series': series,
+            'subtitles': subtitles,
+            'title': title,
+            'upload_date': unified_strdate(air_date),
+        }
```

Comment:
```suggestion
        meta = self._search_json(r'window\._PBS_KIDS_DEEPLINK\s*=', webpage, 'video info')
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
            traverse_obj(meta, ('video_obj', 'URI', {url_or_none})), video_id, ext='mp4')

        return {
            'id': video_id,
            'formats': formats,
            'subtitles': subtitles,
            **traverse_obj(meta, {
                'categories': ('video_obj', 'video_type', {str}, {lambda x: [x]}),
                'channel': 'show_slug',
                'description': ('video_obj', 'description', {str}),
                'duration': ('video_obj', 'duration', {int_or_none}),
                'series': ('video_obj', 'program_title', {str}),
                'title': ('video_obj', 'title', {str}),
                'upload_date': ('video_obj', 'air_date', {unified_strdate}),
            })
        }
```

Reply:
```py
                'categories': ('video_obj', 'video_type', {str}, {lambda x: [x]}),
```
this could return a value of `[None]` because the transformation is always called
so maybe we should do:
```py
                'categories': ('video_obj', 'video_type', {str}, {lambda x: [x] if x else None}),
```

Reply:
@snixon Pls see https://github.com/yt-dlp/yt-dlp/pull/7602#discussion_r1264598023